### ISSUE TITLE
Build noise from deprecated features when compiling against OpenSSL 3

### ISSUE THREAD

**Issue by azdagron** [google/go-tpm-tools#155]
When cgo picks up OpenSSL 3.0 (for example via brew on macOS), compilation outputs quite a few deprecation warnings similar to the following:

```In file included from include.c:172:
/redacted/go/pkg/mod/github.com/google/go-tpm-tools@v0.3.1/simulator/ms-tpm-20-ref/TPMCmd/tpm/src/crypt/ossl/TpmToOsslMath.c:600:9: warning: 'EC_POINTs_mul' is deprecated [-Wdeprecated-declarations]
/usr/local/opt/openssl/include/openssl/ec.h:857:1: note: 'EC_POINTs_mul' has been explicitly marked deprecated here
/usr/local/opt/openssl/include/openssl/macros.h:182:49: note: expanded from macro 'OSSL_DEPRECATEDIN_3_0'
/usr/local/opt/openssl/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
```

One route is to disable deprecation warnings, assuming openssl 3.0 support is off the table for now.

**Comment by josephlr** [comment]
We could manually patch the simulator code, the problem is that the TCG spec explicitly uses this function:
  - https://trustedcomputinggroup.org/resource/tpm-library-specification/
  - Specifically Line `442`, Section `B.2.3.2.3.17.` of [Version 1.59, Part 4 of the TPM 2.0 Spec](https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part4_SuppRoutines_code_pub.pdf)

This means the the C libraries (which conform to the spec) also use this deprecated function:
  - [Call in `libtpms`](https://github.com/stefanberger/libtpms/blob/26e0eea09378628b5cfe789bc4bd9917014e9c26/src/tpm2/crypto/openssl/TpmToOsslMath.c#L699)
  - [Call in `ms-tpm-20-ref`](https://github.com/microsoft/ms-tpm-20-ref/blob/e3c585d7113bae31924d6fd483e65a166a32098f/TPMCmd/tpm/src/crypt/ossl/TpmToOsslMath.c#L600)

I think it would be reasonable to just add a `-Wno-deprecated-declarations` flag to `internal_cgo.go`.

Also, see https://github.com/stefanberger/libtpms/issues/215 for OpenSSL 3.0 support progress in `libtpms`. We will likely also need to remove library use of deprecated functions if we ever want this to build w/ OpenSSL.

**Comment by azdagron** [comment]
Looks like this was fixed recently in #244. Anybody know when the next release is?
